nerdlets/network-telemetry-overview/ip-address.jsx

```diff
@@ -0,0 +1,56 @@
+import { Toast, navigation } from "nr1";
+import PropTypes from "prop-types";
+import React from "react";
+
+import findRelatedAccountsWith from "../../src/lib/find-related-account-with";
+import nrdbQuery from "../../src/lib/nrdb-query";
+
+/**
+ * render an ip address that on click will attempt to find the
+ * host associated with it. Look at NetworkSample event data
+ * and search across all accounts.
+ */
+export default class IpAddress extends React.Component {
+  static propTypes = {
+    value: PropTypes.string,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.onClick = this.onClick.bind(this);
+  }
+
+  async onClick() {
+    const ipAddress = this.props.value;
+    if (!ipAddress) return;
+
+    this.setState({ searching: true });
+    const where = `ipV4Address like '${ipAddress}/%'`;
+    const accounts = await findRelatedAccountsWith({ eventType: "NetworkSample", where });
+    if (accounts.length === 0) {
+      Toast.showToast({
+        description: `Could not find ${ipAddress} on any monitored hosts.`,
+        title: "IP Address Not Found",
+        type: Toast.TYPE.NORMAL,
+      });
+    } else {
+      const account = accounts[0];
+      const nrql = `SELECT latest(entityGuid) as entityGuid FROM NetworkSample WHERE ${where} SINCE 2 minutes ago`;
+      const results = await nrdbQuery(account.id, nrql);
+      const { entityGuid } = results[0];
+      navigation.openStackedEntity(entityGuid);
```

Comment:
hmm this is optimistic. We could easily not get an entityGuid etc. I guess `navigation.openStackedEntity` will silently fail, but maybe should render the same "couldn't find entity" toast if `entityGuid` isnt found...

Reply:
Fixed in latest commit.